### PR TITLE
create demographics group and alive status on person create

### PR DESCRIPTION
**Pull request by TristanB17**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187563276

# A brief description of the changes

Current behavior: person and consumer role are created simultaneously during sign up + adding a family member

New behavior: demographics group and alive status are now generated after a consumer role is generated for a person

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/domain/operations/families/create_member.rb [hunk 1]
```diff
@@ -55,6 +55,7 @@ def persist_person(person)
         vlp_document = person.consumer_role.vlp_documents.first
         consumer_role.active_vlp_document_id = vlp_document.id if vlp_document.present?
         person.save!
+        person.create_demographics_group
```
Comment (vkghub) [inline app/domain/operations/families/create_member.rb:58]: adding here will update person record again. Lets move this to the operation: **app/domain/operations/people/transform_applicant_to_member.rb** and use the pattern present in this operation.

File: app/models/person.rb [hunk 2]
```diff
@@ -1373,6 +1373,24 @@ def alive_status
...
+  # 2.) Family -> build_consumer_role
+  #   - used when creating a family_member from the 'Manage Family' from a user's homepage
+  # 3.) Operations::Families::CreateMember -> persist_person
+  #   - used when an applicant is propagated to a family_member
+  # 4.) Insured::FamiliesHelper -> build_consumer_role
+  #   - used as a failsafe in case any family_members are missing a consumer role
+  # 5.) Operations::People::CreateOrUpdateConsumerRole -> create_consumer_role
+  #   - used when the legacy ::FinancialAssistance::Operations::Families::CreateOrUpdateMember propagates an applicant to a family_member
+  def create_demographics_group
+    return unless demographics_group.blank?
```
Comment (vkghub) [inline app/models/person.rb:1389]: Sine this feature is currently only for consumer role, it is better to add one more check to verify if the person has consumer role.

**Review by vkghub — CHANGES_REQUESTED** [review]

File: app/models/person.rb [hunk 3]
```diff
@@ -1373,6 +1373,22 @@ def alive_status
...
+  #   - used for consumer sign-up
+  # 2.) Family -> build_consumer_role
+  #   - used when creating a family_member from the 'Manage Family' from a user's homepage
+  # 3.) Insured::FamiliesHelper -> build_consumer_role
+  #   - used as a failsafe in case any family_members are missing a consumer role
+  # 4.) Operations::People::CreateOrUpdateConsumerRole -> create_consumer_role
+  #   - used when the legacy ::FinancialAssistance::Operations::Families::CreateOrUpdateMember propagates an applicant to a family_member
+  def create_demographics_group
+    return unless demographics_group.blank? && consumer_role.present?
+    demographics_group = DemographicsGroup.new(alive_status: AliveStatus.new)
```
Comment (ymhari) [inline app/models/person.rb:1388]: Create alive status in a different method, and check the presence of the alive status before create a new one.

**Review by ymhari — CHANGES_REQUESTED** [review]

**Review by vkghub — APPROVED** [review]
LGTM

**Review by ymhari — APPROVED** [review]

**Pull request merged**